golang/cosmos/x/vlocalchain/vlocalchain.go

```diff
@@ -47,7 +49,7 @@ func (h portHandler) Receive(cctx context.Context, str string) (ret string, err
 	case "VLOCALCHAIN_QUERY_MANY":
 		// Copy the JSON messages string into a CosmosTx object so we can
 		// deserialize it with just proto3 JSON.
-		cosmosTxBz := []byte(`{"messages":` + string(msg.Messages) + `}`)
+		cosmosTxBz := bytes.Join([][]byte{[]byte(`{"messages":`), msg.Messages, []byte(`}`)}, []byte{})
```

Comment:
If here (and below) is worth an optimisation, maybe it would be worthwhile to lift the reused `[]byte` arguments to the top of the file?  If you choose to do this, use what you think would be appropriate identifiers.

```go
var (
	messagesObjStart = []byte(`{"messages":`)
	objEnd = []byte(`}`)
)
```


```suggestion
		cosmosTxBz := bytes.Join([][]byte{messagesObjStart, msg.Messages, objEnd}, []byte{})
```

Reply:
Yeah this one was debatable. I later learned that `"foo" + string(bar) + "baz"` does not copy bar before concatenating. And I'm unsure if golang is actually able to avoid the intermediate allocations when concatenating multiple strings. That said, the outer `[]byte()` still does a copy of the concatenated string, so that's worth optimizing.